src/java/map/FloatMask.java

```diff
@@ -261,6 +261,63 @@ public FloatMask add(float val) {
         return this;
     }
 
+    public FloatMask addWeighted(FloatMask other, float weight) {
+        checkSize(other);
+        for (int y = 0; y < getSize(); y++) {
+            for (int x = 0; x < getSize(); x++) {
+                addValueAt(x, y, other.getValueAt(x, y) * weight);
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask addTorroidally(FloatMask other) {
```

Comment:
Sometimes Torroidally has 2 rs and other times only 2

Reply:
fixed